tests/driver_aip31068/main.c

```diff
@@ -410,11 +519,124 @@ int run_demo(int argc, char **argv)
 
     aip31068_clear(&aip31068_dev);
 
+    /* 9. progress bar */
+    _set_progress_bar_enabled(&aip31068_dev, true);
+    aip31068_set_cursor_position(&aip31068_dev, 0, 0);
+    aip31068_print(&aip31068_dev, "Progress: ");
+
+    for (int j = 0; j <= 100; j++) {
+
+        aip31068_set_cursor_position(&aip31068_dev, 0, 10);
+
+        char string_rep[6];
+        sprintf(string_rep, "%d %%", j);
+
+        aip31068_print(&aip31068_dev, string_rep);
+
+        _set_progress(&aip31068_dev, j);
+        xtimer_usleep(100 * US_PER_MS);
+    }
+    _set_progress_bar_enabled(&aip31068_dev, false);
+    aip31068_clear(&aip31068_dev);
+
     puts("[DEMO END]");
 
     return 0;
 }
 
+void _set_progress_bar_enabled(aip31068_t *dev, bool enabled)
+{
+    _progress_bar_enabled = enabled;
+
+    if (enabled) {
+        _init_progress_bar(dev, dev->params.row_count - 1);
+    }
+}
+
+void _set_progress_bar_row(aip31068_t *dev, uint8_t row)
+{
+    _progress_bar_row = row;
+}
+
+void _set_progress(aip31068_t *dev, uint8_t progress)
+{
+    if (!_progress_bar_enabled) {
+        return;
+    }
+
+    /* calculate the number of pixel-columns on a single line */
+    int bar_count = dev->params.col_count * 5;
```

Comment:
For all of these I think you could add a define, something like:

```
#define AIP31068_CHAR_PIXEL_ROWS    (5)
```

```suggestion
    int bar_count = dev->params.col_count * AIP31068_CHAR_PIXEL_ROWS;
```

What do you think?

Reply:
I replaced the 5 with a define, but with a define from the test and not from the driver, because I don't think it's needed there.

Reply:
ok!